### PR TITLE
Allow users to deploy from a local directory.

### PR DESCRIPTION
**Pull request by jhowarth**

### CODE REVIEW

**Comment by jhowarth** [comment]
That's what the `remote_file?` function does.  It's remote if it's a URI.

edit: Nevermind I see what you mean.

**Comment by jhowarth** [comment]
@reset I added more validation in last commit.

**Comment by ivey** [comment]
I'm a little twitchy on putting paths into an attr called url. Main concern being a pedant like me putting in a file:/// URL, which (I think) would make chef try and load it off the server's filesystem. Minor concern, feel free to ignore (since that problem exists prior to the PR) ... maybe a fix is to see if the scheme is file: and turn it into a path?

Probably not worth the trouble, just felt like mentioning it.

:+1:

**Comment by jhowarth** [comment]
@ivey I agree with that.  Perhaps it would be better if we have artifact_url, local_path, and something for nexus stuff.  That way it's clear where you're trying to get the artifact.  This way there would be less magic and confusion.

@reset What do you think of that?

edit: The disadvantage to having several is that it makes the cookbooks slightly more complex as you have to handle a local case and a remote case.  Perhaps just renaming it away from artifact_url to something like artifact_location would be better.

**Comment by reset** [comment]
@jhowarth I like `artifact_location`

**Comment by ivey** [comment]
We should keep artifact_url as undocumented and deprecated, but pass it in to artifact_location so existing cookbooks work.

**Comment by jhowarth** [comment]
@reset @ivey Check out a40f9def4d0d0c1524b8a6ab918417d89eba6d1a

**Comment by jhowarth** [comment]
One problem with having both artifact_location and artifact_url is you can't make artifact_location required.

File: providers/deploy.rb [hunk 1]
```diff
@@ -30,12 +31,17 @@
 attr_reader :previous_versions
 
 def load_current_resource
+  # This is to support deprecated attribute artifact_url.
+  if @new_resource.artifact_url
+    @new_resource.artifact_location = @new_resource.artifact_url
```
Comment (reset) [inline providers/deploy.rb:36]: Let's add a `Chef::Log.warn` in here saying this is a deprecated feature

File: resources/deploy.rb [hunk 2]
```diff
@@ -60,3 +61,8 @@ def release_path
 def shared_path
   "#{self.deploy_to}/shared"
 end
+
+# This is to support deprecated attribute artifact_url.
+def artifact_location=(location)
```
Comment (reset) [inline resources/deploy.rb]: a simple assignment like this can be converted into an `attr_writer`

**Comment by jhowarth** [comment]
@reset

**Comment by reset** [comment]
:+1:! Awesome work man

**Comment by jhowarth** [comment]
@reset Cool with that version bump?

**Pull request merged**